### DISCUSSION TITLE
Idle mode - any way to avoid it?

### DISCUSSION THREAD

**Discussion by richsimpson** [wills106/homeassistant-solax-modbus#202] [Q&A]
I am seeing two stages where the inverter isn't great. 

1 - when the battery storage drops to 10% during the day (the minimum) and through winter there will be more of this. The inverter goes to an 'idle mode' and is hopeless. When I want to force charge the battery in cheap energy tariff, the command is ignored as the inverter is not responsive. I find I have to turn it off and on to get it to wake up. Currently dealt with by sending me an alert 5 mins before the cheap energy slot so I can be prepared.

2 - when the battery gets to full or more often 98% then decides its full. When the sun goes in, the battery doesn't always kick in so I end up buying energy in rather than use the battery. Currently dealt with from another alert I have set so I can activate a high load device to 'jump start' the battery.

Anyone got any other solutions? 

I am going to try setting the battery minimum SOC to 11% to see if that makes a difference, probably not though as imagine inverter will still be idle, just 1% sooner in the day.

**Comment by wills106** [comment]
You need to put the Inverter into Forcetime mode before it goes to sleep.
The Inverter semi shuts down mimicking a PV only Inverter.
Edit: The cloud is able to wake an Inverter up when it's in sleep mode, but I have been unable to workout what command it's sending. The Wake Inverter button in the Integration doesn't wake it up.

The Inverter should pull from the battery at any time during the day unless you are using a backup mode.
The Inverter will never pull 100% from the battery. There is a slight delay in the reaction of it trying to match the load. Stable loads are easier for the Inverter to match, but some loads that quickly change like an Induction Hob is trickier.

**Reply by richsimpson** [reply]
> You need to put the Inverter into Forcetime mode before it goes to sleep. 
> 
> The Inverter should pull from the battery at any time during the day unless you are using a backup mode. The Inverter will never pull 100% from the battery. There is a slight delay in the reaction of it trying to match the load. Stable loads are easier for the Inverter to match, but some loads that quickly change like an Induction Hob is trickier.

Cheers,

When you say force time mode, what do you mean? Its the x1G4Hybrid. I have it set to self use mode typically, then when its cheap energy I put to manual mode and force charge.  Do you mean that at say 11% I should automate to do something else (what?)

On the battery not kicking in when its full, I find that when its set to self-use and the battery fills (say from the solar on a good day) sometimes it gets to the point that solar is no longer enough come sunset and the battery not kicking in and so pulling from the grid. It might do that all night at say 300w. If I create a spike of power it will then kick the battery in - I tend to turn the under floor heating on for 20 seconds to create a high load. 

I have been caught out a few times when I wake up to 100% battery and I have been pulling from the grid all night. Especially if I am on holiday and the house is just ticking along on 300w all day and night.

**Reply by richsimpson** [reply]
Just a thought, for the first part (to avoid idle mode), it the inverter is forced into manual mode at when it drops from 11% to 10% and the stop charge/discharge option, would that keep it alive to then receive another command when its cheap energy? Probably just need to test that to see I guess.

**Reply by richsimpson** [reply]
Bit of testing around the current battery levels and it seems to work. 

I checked my logs and when the battery hit 10% the other afternoon, (Solar should have produced 10 on the forecast but achieved 1! ) the battery level hit 10% at 17:34hrs43secs and the inverter went to idle mode straight away (within 30 seconds or 2 poll updates) at 17:35hrs13secs.

I have just tested this logic by checking for battery dropping from 73% to 72% and forcing inverter to manual mode and to "Stop Charge & Discharge" or Mode 2 in my world. It sat there fine for a good 15 minute in "normal mode" and didn't go to "idle mode". 

![image](https://user-images.githubusercontent.com/51749826/201892479-fac116c9-b16d-464a-b847-88000febf0ae.png)

I have updated my automation so that when the battery gets to 12% the flow starts and when it reaches 11% it will now enter "manual mode/stop charge&discharge". Hopefully this works. 
I have retained the notifications to alert me if this doesn't work though, so 5 mins into my cheap energy window, if the battery hasn't reached 12% I will get another alert to tell me something is wrong. 

The other alert is to warn me if the inverter is in IDLE mode or if it is LOCKED.

**Reply by wills106** [reply]
The Gen4 is a bit different to my Gen3 have you tried enabling "Selfuse Night Charge Enable" so it charges during your timed slot?

If your Inverter is getting stuck at 100% it might be worth seeing if there are any Firmware updates.

**Reply by richsimpson** [reply]
> have you tried enabling "Selfuse Night Charge Enable" so it charges during your timed slot?
> 
> If your Inverter is getting stuck at 100% it might be worth seeing if there are any Firmware updates.

The timed slot seems to cause issues in the summer where the battery needn't charge but does due to the times. So I set them to 2030hrs start and 2030hrs end. Might dabble again over winter as its going to need charging most days. 

I think its a firmware update thats caused the 100% issue. fix one thing and break another. First month or two my inverter wouldnt accept more than 3.7kwh of solar. A firmware update released that and another one has been added to cure the battery sticking at 10% issue but not sure it worked..

My workarounds are working for now. Past few nights it has done the same thing, but the automation to jumpstart has been working.

I will report both issues to solax again and keep my temporary fix until I know they sort it.

**Comment by mrpohoda** [comment]
I was struggling with idle mode too. SolaX Battery Awaken button and a small delay works for me. I have gen3.

```
- device_id: c240ba01535e051efee1df0507ebe650
  domain: button
  entity_id: button.solax_battery_awaken
  type: press
- delay:
    hours: 0
    minutes: 0
    seconds: 15
    milliseconds: 0
- device_id: c240ba01535e051efee1df0507ebe650
  domain: select
  entity_id: select.solax_charger_use_mode
  type: select_option
  option: Force Time Use
```

**Reply by wills106** [reply]
How reliable do you find this? Can you wake the Inverter back up after several hours of it being asleep?

**Reply by mrpohoda** [reply]
This works for me with 100% reliability. At the beginning when I did not use the timeout and the battery awaken button, it almost never worked - I changed the mode to Force time or different, but it automatically changed back after couple of seconds. We use electricity for heating so at this time, our consumption is quite high. I fully charge the battery once during a night when the price is lowest, and once during a day when the price is also good. I've attached history of battery SOC - it is visible that sometimes the battery SOC is at minimum (11%) for more then 12 hours. Every battery charging is done by the automation, which I sent before.

<img width="1179" alt="Screenshot 2022-11-24 at 21 02 37" src="https://user-images.githubusercontent.com/922916/203857527-01ed1dd0-db5e-4623-930d-5489b6913562.png">

**Reply by wills106** [reply]
Thanks for the feedback / investigation work.

It's been annoying not knowing how to successfully wake the Inverter up. Especially when I am no longer in a situation where my Inverter goes to sleep at night. So I couldn't investigate myself.

**Reply by DoMiNiOuS** [reply]
You find a solution to waking the inverter Will or disable idle altogether? Retrofit 3.7E Gen3 for me.
I woke to same issue this morning, battery not charged and in idle mode with modbus unresponsive to any commands. Had to switch inverter off in the end to get it to respond but I have EPS connected and most of my house switches off when I switch the inverter off so not ideal. Im a noob at modbus and am not using node-red like mrpohoda so no idea where the changes he made are actually edited.
Happy new year by the way.

**Reply by DoMiNiOuS** [reply]
so the power on off write on register 0x1C in system idle works as i can see it on the panel switching on and off.....but 0x56 battery_awaken and everything else fails to write, bit strange

**Reply by mrpohoda** [reply]
@DoMiNiOuS I've attached screenshot of my automation and also a Log history. I've found out that Battery awaken button is not necessary. I simply change the Use mode to Force Time Use, wait for about 25s, and then change the use mode to Force Time Use again. If the inverter is in normal mode, these 2 changes are fine. If the inverter is in Idle mode, the use mode immediately returns to Self mode. SolaX run mode goes to Waiting and then Checking mode after the first change. The second change of use mode after 25s is permanent and does not return back to Self use.

<img width="350" alt="Screenshot 2022-12-31 at 8 55 56" src="https://user-images.githubusercontent.com/922916/210129812-63802650-5970-44d1-b7a5-a840041870a6.png">
<img width="541" alt="Screenshot 2022-12-31 at 8 44 11" src="https://user-images.githubusercontent.com/922916/210129815-0edb3f1b-6288-4718-98a1-703e40c6d426.png">

**Reply by DoMiNiOuS** [reply]
Thanks, as I am new to HA as well, couple of things I don't understand...
What are your triggers and what are they looking for I don't have OTE sensor? 
What is the Test if 1 condition.
I matched the battery level condition to my minimum soc if that's right.
Actions I added ok 👍

**Comment by PatrikTrestik** [comment]
I have to add mi finding to this. I experimented with my G4 in Idle mode. It looks like it does not change anything, but at and I found out that value was remembered and set immediately when Inverter goes out of Idle mode.

This is how it works for me:
- Inverter in Idle mode
- Set "Minimum SOC" 50% >> nothing happened and SOC shows 10%
- PV produces some energy or some other event woke inverter. Inverter goes to Normal mode.
- Minimum SOC hops to 50%. Remembered previous setting.

I hope it might be usable for someone.

**Comment by 02curls** [comment]
nothing worked for me to force wake from idle in Self Use Mode, so now in winter I just switched to Backup Mode and I can finally schedule and automate charging as I wish as Backup Mode does not allow the battery to enter Idle mode

**Reply by richsimpson** [reply]
Is that right? So better to switch to backup mode instead of ‘Hold Charge’ when the battery gets low? 

Good time to test as battery was getting low so I have set it to shift to backup mode at 11% SOC to then see what happens. Once it made it to 10% though, the inverter is now in idle mode. I will see if it is still operable from the integration when 2330 comes along and it should try to go back to force charge mode.

UPDATE:
Nope, was stuck in idle so had to go and control it directly to force it back out. Back to the drawing board. For now my solution seems ok. Watch for battery moving from 12% to 11% then set as Manual/Stop Charge Discharge. Just need to keep an eye on if it time out of the battery drops to 10% for some reason.

**Comment by richsimpson** [comment]
Its that time of year again where my battery will deplete and I get the annoying idle mode. More so this year as higher Electricity use so the battery depletes faster.

I had been putting the battery into MANUAL MODE and STOP CHARGE & DISCHARGE when the battery got to 11%, to effectively "Hold the charge" and pull from the grid until my desired charge schedule. But if it gets there early enough then it still drops to 10% and hits the Idle Mode. I guess the inverter continues to pull from the battery to run itself? 

I have a work around that detects battery at 15% and if it is earlier in the day I reduce the power output (slow the discharge rate) and it will go to the HOLD CHARGE routine at a higher Battery SOC %, seems the inverter being on still flattens the battery over time. So I am back looking for other solutions again. 
My charge cycle starts at 2330hrs so if the battery hits 15%
before 2000hrs = reduce charge rate to 1A then go to the HOLD CHARGE position at 15% SOC;
2001-2100hrs = reduce charge rate to 1A then go to the HOLD CHARGE position at 13% SOC;
2101-2200hrs = reduce charge rate to 1A then go to the HOLD CHARGE position at 12% SOC.

@mrpohoda is your solution still going strong? I didn't try it but don't really follow as I have different modes I think. I have a X1 Hybrid G4. You mention Force Time Use, wait 25 seconds then Force Time Use again. What does that mode do so I can work out if I have similar? 

My Inverter runs in Self Use Mode usually then is change to MANUAL MODE and the sub settings of FORCE CHARGE, FORCE EXPORT or STOP CHARGE & DISCHARGE. 
Charge window is effectively off. 2300-2300hrs
Discharge window is effectively always 0000-2359hrs

![image](https://github.com/wills106/homeassistant-solax-modbus/assets/51749826/324068ff-75c0-4aec-bb8d-034b112510bf)

Sure there is an easier way to avoid this idle mode issue. 

Extra thought...
As my Export rate is higher now than off peak import rate (15p export vs 7.5p off peak import), I am ok with charging the batteries all year round now to 100% daily during off peak. Perhaps I should now just leave the inverter in Self Use Mode always, and set the charge window to be 2330-0530 which is the off peak tariff. Would the inverter go idle at 10% still, I guess yes. If it does, would it wake up anyway at 2330hrs and charge because it knows to do that from the built in memory, not me trying to talk to it over modbus? I am going to try that tonight. 

The only down side I see would be having less control in the middle of summer where having the oversized PV array providing more than the inverter 3.7kw value allows me to take perhaps 6kw in and push some to grid and some to battery/home.

**Comment by PatrikTrestik** [comment]
Hi guys. When I use solaxcloud app, it is able to wake inverter from idle mode.
From that I have impression that there is possibility to wake up inverter.
I experimented with it but i have no positive result.

Please can you confirm that when inverter is in idle mode and you set something through solax claud app, it wakes?

**Reply by richsimpson** [reply]
I can turn the inverter off and then back on using the solax app. Then catch it during the coming on phase (checking, waiting,) to set the manual/force charge. That will let me get back to a higher battery SOC then I can hold the state again in Manual/Stop Charge & Discharge. 

I can;t force out of Idle without turning off and on though. So I get an alert that inverter is idle (Hass/Node Red automation) so I can manually go on to the app. Not the best solution.

**Comment by richsimpson** [comment]
Well good news for me last night. Disabled all my automations in Node Red that used to control things to avoid **_IDLE MODE_**. Inverter in effectively factory defaults mode. **_SELF USE MODE_**. Once it hits the minimum battery SOC - what will happen....

![image](https://github.com/wills106/homeassistant-solax-modbus/assets/51749826/6e64d945-754f-432d-957f-252309fefc06)
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/51749826/2ee08b30-4a3d-4cf3-8cd6-4bd551492b9f)

Well The battery got to 10%, inverter hit **_IDLE MODE_** (18/12 23:00hrs)
Set itself into **_STOP CHARGE & DISCHARGE_** sub state of **_MANUAL MODE_** (But doesn't change to Manual Mode so no impact?)
Sits and wait until the 2330hrs **_CHARGER START TIME 1_**
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/51749826/ef4bb978-2cdd-4d95-9dde-d9f795a4d17c)

Then appears to restart, going through the **_CHECKING MODE_** (23:30:06), Enters **_NORMAL MODE_** (23:31:07) a minute later which is the usual restart time on inverter. Then the log is clear until 03:42hrs when I see the battery hit 100% and the Inverter enters **_STANDBY_** Run Mode. Through to just shy of 05:30hrs when the **_CHARGER END TIME 1_** occurs and so Inverter delivers power through to the house again from the storage battery.

So my answer is go back to the default modes. What ever wasn't working for me or my use case before, does work now. If in future I find it is more economical (through more storage capacity, change of use, change of tariff) I will come back. for now tough my automation page is vastly reduced!

**Comment by piotr-zakrzewski** [comment]
Hi there. Is anyone knows WHY System ON using SolaxCloud works but System ON by modbusRTU to not works -> do not escape form IDLE?? Yes Register System ON / OFF change to 1 or 0 by modbus, but that all. No any affect that helps escape from IDLE status. Is there any other register for apply command? I noticed that maybe "Save" in SolaxCloud send any extra command to inverter... Please help.

**Reply by PatrikTrestik** [reply]
I can probably answer "why" but I don't know fix. 

Why? SolaxClaud does not use Modbus at all. It has some API and MQTT which is not accessible to us. 
Modbus is not implemented 1:1 and can work differently.

Fix? We have to find out if any Modbus write can do it. I have SolaxCloud app decompiled and sometimes there is hidden hint. I'll have a look.

**Reply by piotr-zakrzewski** [reply]
In modbus protocol there is a System ON/OFf register. It seems looks like Solax cloud Remote Settings > System On/Off but using Modbus by RS485 inverter do not escape from Idle status. Maybe app send more commands like "Apply changes"? Thanks for Your help.

**Reply by piotr-zakrzewski** [reply]
[PatrikTrestik](https://github.com/PatrikTrestik) - any updates from You side? Maybe you find something interesting in decompiled SolaxCloud app??

**Reply by PatrikTrestik** [reply]
> [PatrikTrestik](https://github.com/PatrikTrestik) - any updates from You side? Maybe you find something interesting in decompiled SolaxCloud app??

Unfotunately nothing of any valuie. As I mentioned before App/Cloud is not using Modbus at all. 
I found no corellation between app values and Modbus. I can confrm that for "System ON", app saves only one value, nothing more.
But it stil does not mean that API does not add some magic to it.

I suggest that someone with modbus skils opens case with SolaX support and asks.
I did it two times, it is long journey to get to reall enginer with skills to answer, but I always get help.

**Reply by romankysely** [reply]
I will do it and let you know.

**Reply by piotr-zakrzewski** [reply]
Hi all. Thanks for all replay. In last week we update SW ARM from version 1.31-1.06 to 1.47-1.06 and DSP version 1.33 to 1.50. After that Command "System On" works by Modbus Protocol like SolaxCloud to "escape" from Idle Status. Thx 🥳

**Reply by romankysely** [reply]
> Hi all. Thanks for all replay. In last week we update SW ARM from version 1.31-1.06 to 1.47-1.06 and DSP version 1.33 to 1.50. After that Command "System On" works by Modbus Protocol like SolaxCloud to "escape" from Idle Status. Thx 🥳

That is very good news. Thanks for info! I am stopping requets to Solax Support. They did not were sure at v1.50 but confirmed that 1.47 does not suport it. Another solution is to use Backup Mode instead Self Use Mode. It does not use Idle Mode.
